owlkettle/customPragmas.nim

```diff
@@ -0,0 +1,8 @@
+## Definess a custom pragma called "locker".
+## This does nothing when compiled with a nim version 2.0 or higher,
+## but applies the "locks: 0" pragma if compiled for e.g. 1.6.X.
+## "locks: 0" is applied to ensure that this code never accesses locked data in user-defined procs.
+when NimMajor >= 2:
+  {.pragma: locker.}
+else:
+  {.pragma: locker, locks: 0.}
```

Comment:
I think this would nicely fit into `common.nim`

Reply:
Based on your lower comment I assume you want to move this into a `customPragma` template and use that in the various modules?

Reply:
Oh, sorry, no, I meant putting the `=destroy` in a template for supporting both 1.6.x and 2.0.0. But if a template is need here, using one is also fine.